### PR TITLE
[Merged by Bors] - chore(Geometry): replace all remaining uses of refine', when possible

### PR DESCRIPTION
**Pull request by grunweg**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
@adomani You might find this interesting: at least in `Geometry`, most remaining `refine`s were actually removable.

**Comment by adomani** [comment]
!bench

**Comment by adomani** [comment]
I wonder whether I should look into robustifying the linter for the replacement of `refine'` into a more general replacement tool, so that it would have caught these `refine'`s as well.

However, I'm also aware that Mario is working on one and his tool will be better.

File: Mathlib/Geometry/Manifold/ContMDiffMFDeriv.lean [hunk 1]
```diff
@@ -135,11 +135,11 @@ protected theorem ContMDiffAt.mfderiv {x₀ : N} (f : N → M → M') (g : N →
...
       rw [(extChartAt I (g x₂)).left_inv hx, (extChartAt I' (f x₂ (g x₂))).left_inv h2x]
-    refine' Filter.EventuallyEq.fderivWithin_eq_nhds _
-    refine' eventually_of_mem (inter_mem _ _) this
+    refine Filter.EventuallyEq.fderivWithin_eq_nhds ?_
+    refine eventually_of_mem (inter_mem ?_ ?_) this
     · exact extChartAt_preimage_mem_nhds' _ hx₂ (extChartAt_source_mem_nhds I (g x₂))
-    refine' extChartAt_preimage_mem_nhds' _ hx₂ _
-    exact h2x₂.continuousAt.preimage_mem_nhds (extChartAt_source_mem_nhds _ _)
+    · refine extChartAt_preimage_mem_nhds' _ hx₂ ?_
+      exact h2x₂.continuousAt.preimage_mem_nhds (extChartAt_source_mem_nhds _ _)
```
Comment (adomani) [inline Mathlib/Geometry/Manifold/ContMDiffMFDeriv.lean:142]: Just flagging an extra `·`, I like it, though!

**Review by adomani — APPROVED** [review]
Modulo benchmarking and building, this looks good to me!

maintainer merge

File: Mathlib/Geometry/Manifold/VectorBundle/SmoothSection.lean [hunk 2]
```diff
@@ -165,9 +165,9 @@ instance instSMul : SMul 𝕜 Cₛ^n⟮I; F, V⟯ := by
   have hs := s.contMDiff x₀
   rw [contMDiffAt_section] at hs ⊢
   set e := trivializationAt F V x₀
-  refine' (contMDiffAt_const.smul hs).congr_of_eventuallyEq _
+  refine' (contMDiffAt_const.smul hs).congr_of_eventuallyEq ?_
   · exact c
```
Comment (sgouezel) [inline Mathlib/Geometry/Manifold/VectorBundle/SmoothSection.lean:169]:
```suggestion
  refine ((contMDiffAt_const (c := c)).smul hs).congr_of_eventuallyEq ?_
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

**Comment by grunweg** [comment]
Thank you for the fast review!
bors r+

**Pull request closed without merging**